engine/src/main/java/org/hibernate/search/annotations/Facet.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.annotations;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Extension annotation for {@code @Field} enabling it for faceting.
+ *
+ * @author Hardy Ferentschik
+ * @hsearch.experimental : This feature is experimental
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.METHOD, ElementType.FIELD })
+@Documented
+public @interface Facet {
+
+	/**
+	 * @return the facet name. Defaults to the name of the field this facet refers to
+	 */
+	String name() default "";
+
+	/**
+	 * @return the field name this annotation refers to. It can be omitted in case there is only a single {@code @Field}
+	 * annotation
+	 */
+	String forField() default "";
+
+	/**
+	 * @return the encoding type to use for this facet. Per default the encoding type is chosen based on the type of the
+	 * entity property.
+	 */
+	FacetEncodingType encoding() default FacetEncodingType.AUTO;
```

Comment:
> Not sure how easy it is, but couldn't our implementation handle discrete facet queries in case the facet field is numeric (via a collector)

Sure we can, but this is what we want to get away from, right? Our facet implementation so far was based on a custom collector. It basically did what you suggest and to a certain degree it was much more flexible. It could basically "on the fly" handle discrete vs range facets. After all it was just loading (collecting) all possible values. The question and rub in our case is how to do this efficiently. The answer so far (and pre Lucene 5) was `FieldCache`s , but again this is something which we want and have to get away from. 

IMO there is no problem with exposing this encoding type. In fact I find it consistent with the latest changes we did for general indexing of numbers. Similarly numbers are now numerically encodied by default and one can force a string encoding by choosing an appropriate bridge. This is really the same thing. 

We have so far not been able to abstract the choice  of string vs numeric encoding away from the user and I am not sure we even can w/o restricting some use cases. 

> In the end, discrete values can be considered as a list of ranges of size 1.

It could be a set of discrete floating point values as well.